repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -75,6 +77,7 @@ import scala.io.Source
   */
 class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode, desiredOutputStream: ServletOutputStream)
   extends javax.servlet.http.HttpServletResponseWrapper(originalResponse) with HeaderInteractor {
+  private val headerWarningLogger = LoggerFactory.getLogger(classOf[HttpServletResponseWrapper].getName + "_headerWarning")
```

Comment:
Well sure, it indicates logger level, that's true. Logger level does not need to directly correlate with the fully-qualified class name, though. That is, I would consider `headerWarning` to be a descendent of the class logger. After all, if I wanted to, say, turn off the class logger, I probably wouldn't want this logger to continue logging. Sure, it is distinct from the class logger, but it is still a logger pertaining to the class. The only other concern I can think of is that you will run into logger name collisions, but in what case would that happen? Other classes with loggers based on the fully-qualified class name should never collide following Java naming conventions. So then you are only concerned about loggers for this class, which you have complete control over (and thus, can provide collisions with). And even at a lower level, you can still fully customize this logger (although you may have to disable `additivity`, but that is always the case with nested loggers in Log4j2).